viper.go

```diff
@@ -1572,6 +1591,10 @@ func (v *Viper) WatchRemoteConfigOnChannel() error {
 }
 
 func (v *Viper) insensitiviseMaps() {
+	if v.caseSensitive {
```

Comment:
This looks unnecessary, at least I think you will never reach this check, since you do it before calling this method:

https://github.com/spf13/viper/pull/592/files#diff-38f1012180f40493dcf780c312a40ec3R1407

I think it would be better to do these checks outside of the `insensitive` methods.